cli/src/action/mod.rs

```diff
@@ -134,26 +137,21 @@ fn msg_from_io_error(err: IoError) -> String {
 #[cfg(feature = "splinter-cli-jwt")]
 // build a signed json web token using the private key
 fn create_cylinder_jwt_auth(key_name: Option<&str>) -> Result<String, CliError> {
-    let default_key_path = dirs::home_dir()
-        .map(|mut p| {
-            p.push(".splinter/keys");
-            p
-        })
-        .ok_or_else(|| CliError::EnvironmentError("Home directory not found".into()))?;
-    let default_path_string = {
-        if let Some(path) = default_key_path.to_str() {
-            path
+    let private_key = if let Some(key_name) = key_name {
+        if key_name.contains('/') {
+            load_key_from_path(Path::new(key_name))
+                .map_err(|err| CliError::ActionError(format!("Unable to retrieve key: {}", err)))?
         } else {
-            return Err(CliError::ActionError(
-                "Path is not valid unicode".to_string(),
-            ));
+            load_key(key_name, &current_user_search_path())
+                .map_err(|err| CliError::ActionError(format!("Unable to retrieve key: {}", err)))?
+                .ok_or_else(|| CliError::ActionError("Key not found".to_string()))?
         }
+    } else {
+        load_key(&current_user_key_name(), &current_user_search_path())
```

Comment:
This needs to add ~/.splinter/keys to the search path to be consistent with keygen

Reply:
Fixed